hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -141,17 +157,18 @@ withDirectChain tracer networkMagic iocp socketPath keyPair party cardanoKeys ca
                 { postTx = \tx -> do
                     traceWith tracer $ ToPost tx
 
-                    res <- atomically $ do
-                      fromPostChainTx wallet headState cardanoKeys tx >>= \case
-                        Nothing ->
-                          pure Nothing
-                        Just partialTx ->
-                          Just <$> finalizeTx wallet headState partialTx
+                    res <- timeout 10 $
```

Comment:
We introduced this as the `coverFee` is `retry`ing when it cannot find a suitable input to pay fees. The `retry` is necessary because we might have not seen and applied a block giving us the utxo back -> a consequence of two distinct connections to the same `cardano-node`. I believe I speak for @abailly-iohk if I say we are not happy about this either.